app/models/user.rb

```
@@ -17,6 +18,38 @@ def email_with_type
     "#{email}#{type_if_consultant}"
   end
 
+  def set_user_code_if_necessary
+    # if user has already logged in, rely on guid to link (and not user code)
+    if guid
+      user_code_destroy
+    else
+      set_user_code
+    end
+  end
+
+  def user_code_destroy
+    return unless user_code
+    self.user_code = nil if user_code.destroy
+  end
+
+  def set_user_code
+    # if user code already set, bump expiration
+    if user_code
+      user_code.set_expiration
+    else
+      create_user_code
+    end
+  end
+
+  def merge_attributes(other_user)
+    other_user.group_memberships.update_all(user_id: id)
+    other_user.reviews_facilitating.update_all(facilitator_id: id)
+    other_user.reviews_started.update_all(initiator_id: id)
+    other_user.reviews_initiated.update_all(initiator_id: id)
+
```

If other_user is a consultant _and_ we are not, fail noisily. Better yet, `other_user.reviews_consulting.update_all(consultant_id: nil)` - I think we discussed this was acceptable.